consensus/istanbul/backend/internal/enodes/val_enode_db.go

```diff
@@ -436,3 +436,22 @@ func (vet *ValidatorEnodeDB) iterateOverAddressEntries(onEntry func(common.Addre
 	}
 	return iter.Error()
 }
+
+type ValEnodeEntryInfo struct {
+	Enode     string `json:"enode"`
+	Timestamp uint   `json:"timestamp"`
+}
+
+func (vet *ValidatorEnodeDB) ValEnodeTableInfo() (map[string]*ValEnodeEntryInfo, error) {
+	valEnodeTableInfo := make(map[string]*ValEnodeEntryInfo)
+
+	valEnodeTable, err := vet.GetAllValEnodes()
```

Comment:
i don't remember, but i suppose getAllVAlEnodes does the locking here.

Reply:
`GetAllValEnodes` does do the locking.

Note that this new function does go against your initial pattern of having all public functions acquire the lock.   Do you see an issue with that?

Reply:
not at all, since it call another public function that acquires the lock.

btw, i'm seeing the initial patter on geth1.9